dbt/include/sqlserver/macros/materializations/snapshot/snapshot.sql

```diff
@@ -1,206 +1,4 @@
-
-      {% set staging_table = build_snapshot_staging_table(strategy, sql, target_relation) %}
-
-      {% do adapter.expand_target_column_types(from_relation=staging_table,
-                                               to_relation=target_relation) %}
-
-      {% set missing_columns = adapter.get_missing_columns(staging_table, target_relation)
-                                   | rejectattr('name', 'equalto', 'dbt_change_type')
-                                   | rejectattr('name', 'equalto', 'DBT_CHANGE_TYPE')
-                                   | rejectattr('name', 'equalto', 'dbt_unique_key')
-                                   | rejectattr('name', 'equalto', 'DBT_UNIQUE_KEY')
-                                   | list %}
-
-      {% do create_columns(target_relation, missing_columns) %}
-
-      {% set source_columns = adapter.get_columns_in_relation(staging_table)
-                                   | rejectattr('name', 'equalto', 'dbt_change_type')
-                                   | rejectattr('name', 'equalto', 'DBT_CHANGE_TYPE')
-                                   | rejectattr('name', 'equalto', 'dbt_unique_key')
-                                   | rejectattr('name', 'equalto', 'DBT_UNIQUE_KEY')
-                                   | list %}
-
-      {% set quoted_source_columns = [] %}
-      {% for column in source_columns %}
-        {% do quoted_source_columns.append(adapter.quote(column.name)) %}
-      {% endfor %}
-
-      {% call statement('main') %}
-          {{ snapshot_merge_sql(
-                target = target_relation,
-                source = staging_table,
-                insert_cols = quoted_source_columns
-             )
-          }}
-          {{ drop_relation(staging_table) }}
-      {% endcall %}
-
-  {% endif %}
-
-  {{ adapter.commit() }}
-
-  {% if staging_table is defined %}
-      {% do post_snapshot(staging_table) %}
-  {% endif %}
-
-{% endmaterialization %}
-
-{% macro sqlserver__post_snapshot(staging_relation) %}
-  -- Clean up the snapshot temp table
-  {% do drop_relation(staging_relation) %}
-{% endmacro %}
-
-
-{% macro build_snapshot_table(strategy, sql) %}
-
-    select *,
-        COALESCE({{ strategy.scd_id }}, NULL)  as dbt_scd_id,
-        COALESCE({{ strategy.updated_at }}, NULL) as dbt_updated_at,
-        COALESCE({{ strategy.updated_at }}, NULL) as dbt_valid_from,
-        nullif({{ strategy.updated_at }}, {{ strategy.updated_at }}) as dbt_valid_to
-    from (
-        {{ sql }}
-    ) sbq
-
-{% endmacro %}
-
-{% macro sqlserver__create_columns(relation, columns) %}
-  {% for column in columns %}
-    {% call statement() %}
-      alter table {{ relation }} add "{{ column.name }}" {{ column.data_type }};
-    {% endcall %}
-  {% endfor %}
-{% endmacro %}
\ No newline at end of file
+{% macro sqlserver__post_snapshot(staging_relation) %}
+  -- Clean up the snapshot temp table
+  {% do drop_relation(staging_relation) %}
+{% endmacro %}
```

Comment:
my understanding is that in removing all these custom snapshots macros, that the default macros are now compatible with `sqlserver`?

Reply:
Yes! Most of them are :-).

Reply:
ok awesome. this definitely makes things simpler for @nandanhegde15 and I with `dbt-synapse`. Right now `MERGE` is in only private preview for Synapse, so we have to hack our own version of the `snapshot_merge_sql()` macro...